Revit_Core_Engine/Convert/Environment/FromRevit/CableTray.cs

```diff
@@ -79,9 +89,58 @@ public static List<BH.oM.MEP.System.CableTray> CableTrayFromRevit(this Autodesk.
                     StartPoint = segment.StartPoint(),
                     EndPoint = segment.EndPoint(),
                     SectionProperty = sectionProperty,
-                    ConnectionProperty = null,
+                    ConnectionProperty = new CableTrayConnectionProperty(),
                     OrientationAngle = orientationAngle
                 };
+                Vector bhomCableTrayVector = BH.Engine.Geometry.Modify.RoundCoordinates((thisSegment.StartPoint - thisSegment.EndPoint),4).Normalise();
```

Comment:
I would consider migrating this whole part into a separate method, particularly in the case it could possibly be reused for other MEP types.

Reply:
You're quite right, but I'd prefer not act on this if you're ok with it. Reason being that the other MEP obj don't take advantage of connector information. Also the only reason I'm doing this is to be more efficient in the path finding, since otherwise I'd have to query this info during it's runtime, so it saves time to grab it during the converts.

Reply:
Makes sense after the 2nd look at the code. Happy to keep it as is then, can be migrated out to a separate method in case of other converts wanting to use it too.